security/fma-message-passing-contracts.md

```diff
@@ -0,0 +1,140 @@
+
+- **Description**: The `relayMessage` in `L2ToL2CrossDomainMessenger` executes a message on the destination chain after `validateMessage` in `CrossL2Inbox`, which is safeguarded from reentrancies by `nonReentrant`. If this function is buggy or poorly implemented, an entity may exploit it to execute a reentrancy attack.
+- **Risk Assessment:** Medium.
+    - Potential Impact: Critical. This could lead to repeated relay of messages. Examples include:
+        - Multiple calls that force execution of `crosschainMint` or `mint` in `SuperchainERC20` and `SuperchainWETH`, respectively. This would break asset invariants, allowing `SuperchainERC20` or ETH assets to be minted improperly, impacting other smart contracts, initiating unauthorized withdrawals, and affecting services such as exchanges and third-party bridges.
+    - Likelihood: Very low. The use of `TransientContext` with reentrancy guards, combined with a careful implementation of the check-effect-interaction pattern in the contract code, makes it robust against such scenarios.
+- **Mitigations:**  Our current codebase includes tests to ensure `relayMessage` reverts when reentrancy is attempted ([test](https://github.com/ethereum-optimism/optimism/blob/66fea80cb9d06a74f0084e1b4fd1a1aa2317b19c/packages/contracts-bedrock/test/L2/L2ToL2CrossDomainMessenger.t.sol#L481)).
+- **Detection:** There should be monitoring tooling that detect repeated calls and event emissions for the same transaction.
+- **Recovery Path(s):** Coordinate an L2 hard fork to revert the subsequent states derived from the failure, and upgrade the predeploy contracts to patch the issue. Pause the system via `SuperchainConfig` if there are dependencies in L1 smart contracts (e.g., a withdrawal from a canonical or custom bridge nearing finalization in `OptimismPortal`). Notify service providers that may be impacted by this failure such as bridges, oracles, and exchanges.
+
+### FM6 : A repeated identifier is validated through `CrossL2Inbox`
+
+- **Description:** The `validateMessage` in `CrossL2Inbox` does not store or track whether a given identifier has been used before. If the same identifier is provided in multiple calls, messages could be executed multiple times, leading to unintended consequences.
+- **Risk Assessment:** Medium.
+    - Potential Impact: High. A `validateMessage` call with a repeated identifier could cause an action to be executed multiple times in contracts that rely on `CrossL2Inbox`, similar to the impact in FM4. However, `L2ToL2CrossDomainMessenger` has replay protection.
+    - Likelihood: Low. Core contracts such as `SuperchainTokenBridge` and `SuperchainWETH` relies on `L2ToL2CrossDomainMessenger`, but other contracts that rely solely on `CrossL2Inbox` could be affected. Even in such cases, chain progression will eventually invalidate repeated calls.
+- **Mitigations:** Ensure proper client-side testing to prevent messages from being validated multiple times.
```

Comment:
Also documentation and communication to devs, similar to the create2 requirement for superchainERC20